scanner/src/main/kotlin/Scanner.kt

```diff
@@ -91,9 +92,10 @@ class Scanner(
 
         scannerWrappers.values.flatten().distinct().forEach { scannerWrapper ->
             scannerWrapper.criteria?.let { criteria ->
-                require(criteria.matches(scannerWrapper.details)) {
-                    "The scanner details of scanner '${scannerWrapper.details.name}' must satisfy the configured " +
-                        "criteria for looking up scan storage entries."
+                val details = with(scannerWrapper) { ScannerDetails(name, version, configuration) }
```

Comment:
It seems a bit tedious to have use this line everywhere the details are needed, is there a reason you didn't add the details as a lazy property to `ScannerWrapper`?

Reply:
It felt like duplicating code to have the properties in there both individually and wrapped in `ScannerDetails`... but as you correctly noted we now have some duplication elsewhere. I'll play with the idea to see how the code would look like.

Reply:
Kotlin says "delegated properties are not allowed on interfaces" 😞

Reply:
I've introduced a convenience extension property now.